File: server-nodejs/src/tools/TinyId.js
export default class TinyId {
	static TINY_ID_CHARACTERS = ["bcdefghijklmnopqrstuvwxyz", "aeiouy"];
	static TINY_ID_LENGTH = 4;

	constructor(tinyId) {
		if (tinyId instanceof TinyId) return new TinyId(tinyId.toString());

		this.id = String(tinyId);
	}

	static generate() {
		let str = "";
		for (let i = 0; i < TinyId.TINY_ID_LENGTH; i++) {
			const array = TinyId.TINY_ID_CHARACTERS[i % TinyId.TINY_ID_CHARACTERS.length];
			str += array[Math.floor(Math.random() * array.length)];
		}

		return new TinyId(str);
	}

	toString() {
		return this.id;
	}
}

File: server-go/tools/tinyId.go
package tools

import (
	"math/rand"
)

var (
	tinyIdCharacters = []([]rune){[]rune("bcdefghijklmnopqrstuvwxyz"), []rune("aeiouy")}
	tinyIdLength     = 4
)

func GenetareTinyId() string {
	str := ""
	tinyIdCharactersLength := len(tinyIdCharacters)
	for i := 0; i < tinyIdLength; i++ {
		array := tinyIdCharacters[i%tinyIdCharactersLength]
		str = str + string(array[rand.Intn(len(array))])
	}

	return str
}
